### PR TITLE
separate segment key for sandbox and devspaces

### PR DESCRIPTION
**Pull request by ranakan19**
fix wrt: https://issues.redhat.com/browse/CRT-1660

Changes in reflect separate segment keys for sandbox and devspaces. DevSpacesAnalytics is now a subset of RegistrationServiceAnalyticsConfig

paired with: https://github.com/codeready-toolchain/toolchain-e2e/pull/733

related PRs: 
https://github.com/codeready-toolchain/api/pull/360, https://github.com/codeready-toolchain/api/pull/361
https://github.com/codeready-toolchain/toolchain-common/pull/313

### CODE REVIEW

File: pkg/controller/woopra.go [hunk 1]
```diff
@@ -18,15 +18,22 @@ func NewWoopra() *Woopra {
 }
 
 // GetHandler returns the woopra-domain for UI.
-func (w *Woopra) GetWoopraDomain(ctx *gin.Context) {
+func (w *Woopra) GetDevSpacesWoopraDomain(ctx *gin.Context) {
```
Comment (rajivnathan) [inline pkg/controller/woopra.go:21]: I guess the comment should be updated to `woopra-domain for DevSpaces`?

File: pkg/controller/woopra_test.go [hunk 2]
```diff
@@ -73,7 +73,7 @@ func (s *TestWoopraSuite) TestWoopraHandler() {
 		req, err := http.NewRequest(http.MethodGet, "/api/v1/segment-write-key", nil)
 		require.NoError(s.T(), err)
 
-		handler := gin.HandlerFunc(woopraCtrl.GetSegmentWriteKey)
+		handler := gin.HandlerFunc(woopraCtrl.GetSandboxSegmentWriteKey)
```
Comment (rajivnathan) [inline pkg/controller/woopra_test.go:76]: I guess we should add tests for GetDevSpacesSegmentWriteKey too?

File: pkg/server/routes.go [hunk 3]
```diff
@@ -39,8 +39,8 @@ func (srv *RegistrationServer) SetupRoutes() error {
 		unsecuredV1 := srv.router.Group("/api/v1")
 		unsecuredV1.GET("/health", healthCheckCtrl.GetHandler)
 		unsecuredV1.GET("/authconfig", authConfigCtrl.GetHandler)
-		unsecuredV1.GET("/woopra-domain", woopraCtrl.GetWoopraDomain)
-		unsecuredV1.GET("/segment-write-key", woopraCtrl.GetSegmentWriteKey)
+		unsecuredV1.GET("/woopra-domain", woopraCtrl.GetDevSpacesWoopraDomain)
+		unsecuredV1.GET("/segment-write-key", woopraCtrl.GetSandboxSegmentWriteKey)
```
Comment (rajivnathan) [inline pkg/server/routes.go:43]: We should add an endpoint for GetDevSpacesSegmentWriteKey too

**Review by rajivnathan — COMMENTED** [review]
Looks good overall, just a few comments

File: pkg/controller/woopra.go (hunk 1 above)
Comment (ranakan19) [reply pkg/controller/woopra.go:21]: done in [3befcfd](https://github.com/codeready-toolchain/registration-service/pull/323/commits/3befcfd9dcd0c7ea9c2515921c69413c5ef1f927)

File: pkg/controller/woopra_test.go (hunk 2 above)
Comment (ranakan19) [reply pkg/controller/woopra_test.go:76]: done in [3befcfd](https://github.com/codeready-toolchain/registration-service/pull/323/commits/3befcfd9dcd0c7ea9c2515921c69413c5ef1f927)

File: pkg/server/routes.go (hunk 3 above)
Comment (ranakan19) [reply pkg/server/routes.go:43]: done in [3befcfd](https://github.com/codeready-toolchain/registration-service/pull/323/commits/3befcfd9dcd0c7ea9c2515921c69413c5ef1f927)

**Comment by ranakan19** [comment]
/retest
made e2e test changes

**Review by xcoulon — APPROVED** [review]

**Review by ibuziuk — APPROVED** [review]

**Pull request merged**